internal/provider/resource_tfe_workspace_run_task_schemas.go

```diff
@@ -0,0 +1,164 @@
+			)),
+			Required: true,
+			Validators: []validator.String{
+				stringvalidator.OneOf(workspaceRunTaskEnforcementLevels()...),
+			},
+		},
+		"stage": schema.StringAttribute{
+			Description: fmt.Sprintf("The stage to run the task in. Valid values are %s.", sentenceList(
+				workspaceRunTaskStages(),
+				"`",
+				"`",
+				"and",
+			)),
+			Optional: true,
+			Computed: true,
+			Default:  stringdefault.StaticString(string(tfe.PostPlan)),
+			Validators: []validator.String{
+				stringvalidator.OneOf(workspaceRunTaskStages()...),
+			},
+		},
+	},
+}
+
+type modelTFEWorkspaceRunTaskV1 struct {
+	ID               types.String `tfsdk:"id"`
+	WorkspaceID      types.String `tfsdk:"workspace_id"`
+	TaskID           types.String `tfsdk:"task_id"`
+	EnforcementLevel types.String `tfsdk:"enforcement_level"`
+	Stage            types.String `tfsdk:"stage"`
+	Stages           types.List   `tfsdk:"stages"`
+}
+
+var resourceWorkspaceRunTaskSchemaV1 = schema.Schema{
+	Version: 1,
+	Attributes: map[string]schema.Attribute{
+		"id": schema.StringAttribute{
+			Computed:    true,
+			Description: "Service-generated identifier for the workspace task",
+			PlanModifiers: []planmodifier.String{
+				stringplanmodifier.UseStateForUnknown(),
+			},
+		},
+		"workspace_id": schema.StringAttribute{
+			Description: "The id of the workspace to associate the Run task to.",
+			Required:    true,
+			PlanModifiers: []planmodifier.String{
+				stringplanmodifier.RequiresReplace(),
+			},
+		},
+		"task_id": schema.StringAttribute{
+			Description: "The id of the Run task to associate to the Workspace.",
+			Required:    true,
+			PlanModifiers: []planmodifier.String{
+				stringplanmodifier.RequiresReplace(),
+			},
+		},
+		"enforcement_level": schema.StringAttribute{
+			Description: fmt.Sprintf("The enforcement level of the task. Valid values are %s.", sentenceList(
+				workspaceRunTaskEnforcementLevels(),
+				"`",
+				"`",
+				"and",
+			)),
+			Required: true,
+			Validators: []validator.String{
+				stringvalidator.OneOf(workspaceRunTaskEnforcementLevels()...),
+			},
+		},
+		"stage": schema.StringAttribute{
+			DeprecationMessage: "stage is deprecated, please use stages instead",
+			Description: fmt.Sprintf("The stage to run the task in. Valid values are %s.", sentenceList(
+				workspaceRunTaskStages(),
+				"`",
+				"`",
+				"and",
+			)),
+			Optional: true,
+			Computed: true,
+			Default:  stringdefault.StaticString(string(tfe.PostPlan)),
+			Validators: []validator.String{
+				stringvalidator.OneOf(workspaceRunTaskStages()...),
+			},
+		},
+		"stages": schema.ListAttribute{
+			ElementType: types.StringType,
+			Description: fmt.Sprintf("The stages to run the task in. Valid values are %s.", sentenceList(
+				workspaceRunTaskStages(),
+				"`",
+				"`",
+				"and",
+			)),
+			// Default: listdefault.StaticValue(basetypes.NewListValueMust(types.StringType, []attr.Value{
+			// 	basetypes.NewStringValue(string(tfe.PostPlan)),
+			// })),
```

Comment:
Should stages have a default? 🤔 If not, we should probably remove this^

Reply:
Good call!  fixed